model-engine/model_engine_server/core/loggers.py

```diff
@@ -35,19 +34,35 @@
     "loggers_at_level",
     # utils
     "filename_wo_ext",
-    "get_request_id",
-    "set_request_id",
+    "LoggerTagKey",
+    "LoggerTagManager",
 )
 
-
-def get_request_id() -> Optional[str]:
-    """Get the request id from the context variable."""
-    return ctx_var_request_id.get()
-
-
-def set_request_id(request_id: str) -> None:
-    """Set the request id in the context variable."""
-    ctx_var_request_id.set(request_id)  # type: ignore
+class LoggerTagKey(str, Enum):
+    REQUEST_ID = 'request_id'
+    TEAM_ID = 'team_id'
+    USER_ID = 'user_id'
+
+class LoggerTagManager:
+    _context_vars: Dict[LoggerTagKey, contextvars.ContextVar] = {}
```

Comment:
wonder if a `defaultdict` works here, although I guess it's probably more effort to get the code into a form where it works with `defaultdict`